src/features/courses/course-controller.ts

```diff
@@ -3362,6 +3363,63 @@ class CourseController {
             return result;
         });
     }
+
+    // Sends an email to the professor associated with the course.
+    async emailProfessor(options: EmailProfOptions): Promise<boolean> {
+        const coursePromise = Course.findOne({
+            where: {
+                id: options.courseId,
+            },
+            include: [{
+                model: User,
+                as: 'instructor',
+                attributes: ['id', 'firstName', 'lastName', 'email', 'preferredEmail'],
+                where: {
+                    active: true
+                },
+                required: true,
+            }],
+        });
+
+        const gradePromise = StudentGrade.findOne({
+            where: {
+                userId: options.student.id,
+                courseWWTopicQuestionId: options.question.id,
+            },
+            attributes: ['randomSeed']
+        });
+
+        const [course, grade] = await Promise.all([coursePromise, gradePromise]);
+
+        if (_.isNil(course) || _.isNil(course.instructor)) {
+            throw new WrappedError('Could not find an instructor for this course.');
+        }
+
+        if (_.isNil(grade)) {
+            throw new WrappedError('Could not find a grade associated with the problem this student is trying to ask about.');
+        }
+
+        const poorMansTemplate = `
+Hello Professor ${course.instructor.lastName},
+
+    Your student ${options.student.firstName} ${options.student.lastName} is asking for help with
+Problem ${options.question.problemNumber+1} (ID#${options.question.id}) in the Topic ${options.topic.name} (ID#${options.topic.id})
+The WeBWorK path for this problem is ${options.question.webworkQuestionPath} and the seed value is ${grade.randomSeed}.
+
+Here is the message that was sent:
+
+${options.content}
+
+You should be able to reply to the student's email address (${options.student.preferredEmail}) by replying to this message.
+`;
+
+        return emailHelper.sendEmail({
+            content: poorMansTemplate,
+            email: course.instructor.preferredEmail,
+            subject: `${options.student.firstName} - Topic ${options.topic.id} - Question ${options.question.id}`,
+            from: options.student.preferredEmail,
```

Comment:
Do we have issues with it detecting as spam? I feel like including a from with an unverified email typically results in instant spam

Reply:
I felt like their was probably a distinction and now I see it: `...(options.from ? {headers: {'Reply-To': options.from }} : undefined)`

I would assume this is good but you should try from other mail platforms other than gmail

Either way I think it is fine for now

Reply:
reply-to is the way to go here

Reply:
sendGrid actually blocks you from using `from` from an unverified domain. ReplyTo works, I just didn't change the name in the option.

Reply:
Right, that's why I was misled, look good though